src/test/extensions/emulator/optionsHelper.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { expect } from "chai";
+import * as sinon from "sinon";
+
+import * as optionsHelper from "../../../extensions/emulator/optionsHelper";
+import { ExtensionSpec } from "../../../extensions/extensionsApi";
+import * as paramHelper from "../../../extensions/paramHelper";
+
+describe("optionsHelper", () => {
+  describe("getParams", () => {
+    const testOptions = {
+      project: "test",
+      testParams: "test.env",
+    };
+    const autoParams = {
+      PROJECT_ID: "test",
+      EXT_INSTANCE_ID: "test",
+      DATABASE_INSTANCE: "test",
+      DATABASE_URL: "https://test.firebaseio.com",
+      STORAGE_BUCKET: "test.appspot.com",
+    };
+    let testSpec: ExtensionSpec;
+    let readParamsFileStub: sinon.SinonStub;
+
+    beforeEach(() => {
+      testSpec = {
+        name: "test",
+        version: "0.1.0",
+        resources: [],
+        sourceUrl: "https://my.stuff.com",
+        params: [],
+      };
+      readParamsFileStub = sinon.stub(paramHelper, "readParamsFile");
+    });
+
+    afterEach(() => {
+      readParamsFileStub.restore();
+    });
+
+    it("should return user and autopopulated params", () => {
+      testSpec.params = [
+        {
+          label: "param1",
+          param: "USER_PARAM1",
+        },
+        {
+          label: "param2",
+          param: "USER_PARAM2",
+        },
+      ];
+      readParamsFileStub.resolves({
+        USER_PARAM1: "val1",
+        USER_PARAM2: "val2",
+      });
+
+      expect(optionsHelper.getParams(testOptions, testSpec)).to.eventually.deep.eq(
+        Object.assign(
```

Comment:
nit* I've grown fond of the spread operator:

```
{
  ...{
            USER_PARAM1: "val1",
            USER_PARAM2: "val2",
          },
  ...autoParams
}
```

Do you prefer `Object.assign`?

Reply:
I'll switch this over - i actually had a bug here for a bit caused by using Object.assign that wouldn;t have happened with a spread operator, which seems like a good enough reason to favor your option.